cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -777,9 +777,15 @@ private String executeWorkflow(ApplicationManager applicationManager, Map<String
                             nodeStateDetailMap.get("WordCount").getRunId());
     Assert.assertEquals(7, getMetricsManager().getTotalMetric(mrMetricsContext, "user.num.words"));
 
-    Map<String, String> readerContext = new HashMap<>(workflowMetricsContext);
+    final Map<String, String> readerContext = new HashMap<>(workflowMetricsContext);
     readerContext.put(Constants.Metrics.Tag.NODE, "readerAction");
-    Assert.assertEquals(6, getMetricsManager().getTotalMetric(readerContext, "user.unique.words"));
+
+    Tasks.waitFor(6L, new Callable<Long>() {
```

Comment:
why do we need retry here? Shouldn't the workflow already be done by this point?